src/odemis/acq/align/fastem.py

```diff
@@ -18,22 +20,254 @@
+DESCAN_GAIN_STATIC = descan_gain
+IMAGE_ROTATION_PREALIGN = image_rotation_pre_align
+IMAGE_TRANSLATION_PREALIGN = image_translation_pre_align
+IMAGE_ROTATION_FINAL = image_rotation
+IMAGE_TRANSLATION_FINAL = image_translation
+
+# The executor is a single object, independent of how many times the module (fastem.py) is loaded.
 _executor = model.CancellableThreadPoolExecutor(max_workers=1)
 
 
-def align(main_data):
+def align(scanner, multibeam, descanner, detector, stage, ccd, beamshift, det_rotator, calibrations):
     """
-    :param main_data (odemis.gui.model.FastEMMainGUIData):
-    :returns: (ProgressiveFuture): acquisition future
+    Start a calibration task for a given list of calibrations.
+
+    :param scanner: (xt_client.Scanner) Scanner component connecting to the XT adapter.
+    :param multibeam: (technolution.EBeamScanner) The multibeam scanner component of the acquisition server module.
+    :param descanner: (technolution.MirrorDescanner) The mirror descanner component of the acquisition server module.
+    :param detector: (technolution.MPPC) The detector object to be used for collecting the image data.
+    :param stage: (actuator.ConvertStage) The stage in the corrected scan coordinate system, the x and y axes are
+        aligned with the x and y axes of the multiprobe and the multibeam scanner.
+    :param ccd: (model.DigitalCamera) A camera object of the diagnostic camera.
+    :param beamshift: (tfsbc.BeamShiftController) Component that controls the beamshift deflection.
+    :param det_rotator: (tmcm.CANController) K-mirror controller.
+    :param calibrations: (list of str) List of calibrations that should be run.
+
+    :returns: (ProgressiveFuture) Alignment future object, which can be cancelled.
+            The result of the future is: (None)
     """
+
+    if not fastem_calibrations:
+        raise ModuleNotFoundError("fastem_calibration module missing. Cannot run calibrations.")
+
     f = model.ProgressiveFuture()
-    _executor.submitf(f, _run_fake_alignment)
+
+    # Create a task that runs the calibration and alignments.
+    task = CalibrationTask(scanner, multibeam, descanner, detector, stage, ccd, beamshift, det_rotator, f, calibrations)
+
+    f.task_canceller = task.cancel  # lets the future know how to cancel the task.
+
+    # Connect the future to the task and run it in a thread.
+    # task.run is executed by the executor and runs as soon as no other task is executed
+    _executor.submitf(f, task.run)
+
     return f
 
 
-def _run_fake_alignment():
-    time.sleep(2)
+def estimate_calibration_time(calibrations):
+    """
+    Computes the approximate time it will take to run all calibrations.
+    :param calibrations: (list of str) List of calibrations that should be run.
+    :return (0 <= float): The estimated time for the requested calibrations in s.
+    """
+    tot_time = 0
+    for calib in calibrations:
+        tot_time += calib.estimate_calibration_time()
```

Comment:
This can be written as `return sum(c.estimate_calibration_time() for c in calibrations)`. Though, this only works if the calibrations are functions. Otherwise, you'll need to use a dict.

Maybe I've done too much code golfing? Actually, I like using `sum()`.